File: problems/C++/62. Unique Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        long long N = m + n - 2;
        long long K = min(m - 1, n - 1);
        long long res = 1;

        for (int i = 0; i < K; ++i) {
            res = res * (N - i) / (i + 1);
        }
        
        return (int)res;
    }
};

File: problems/Python/62. Unique Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        N = m + n - 2
        K = min(m - 1, n - 1)
        res = 1
        
        for i in range(K):
            res = res * (N - i) // (i + 1)
            
        return res